R/checks.R

```diff
@@ -2358,26 +2358,14 @@ checkBadFiles <- function(package_dir){
         return(invisible())
     }
     licenses <- read.dcf(ldb_file)
-    sss <- licenses[, "SSS"]
-    abbrev <- licenses[, "Abbrev"]
-    abbrev[!is.na(sss)] <- sss[!is.na(sss)]
-    restrict <- licenses[, "Restricts_use"]
-    idx <- (restrict == "yes") & !is.na(restrict) & !is.na(abbrev)
-
-    ## PCAN/DESCRIPTION:License: CC BY-NC-ND 4.0
-    ## QUBIC/DESCRIPTION:License: CC BY-NC-ND 4.0 + file LICENSE
-    test0 <- any(vapply(abbrev[idx], grepl, logical(1), license, fixed = TRUE))
-    test1 <-
-        !any(vapply(
-             abbrev[!is.na(abbrev)], grepl, logical(1), license, fixed = TRUE
-         ))
-    if (test0) {
+    result <- tools:::analyze_licenses(license, licenses)
+    test <- result[["restricts_use"]]
+    if (isTRUE(test))
         handleError("License '", license, "' restricts use")
-    } else if (test1) {
+    else if (is.na(test) || !result[, "is_verified"])
         handleNote(
             "License '", license, "' unknown; licenses cannot restrict use"
```

Comment:
I'm not really sure what `is_verified` means, but is the error message saying too much? e.g., if the license *were* known, maybe it would not restrict use?

I'm rusty on these things, but should the handleNote make a reference to a vignette where the nuances of the situation are handled -- e.g., where acceptable licenses are found / when an exceptional license would be acceptable / etc?

Reply:
The documentation is non-existent but the source code seems to indicate that `is_verified` refers to whether the license is for (Free-Open-Source-Software) FOSS. 

I've added a `message` pointing the developer to https://choosealicense.com/appendix and added additional details in the vignette.